### PR TITLE
Add SecretFlow

### PR DESCRIPTION
**Pull request by mkskeller**
This pull requests adds everything necessary to run the examples in [SecretFlow](https://www.secretflow.org.cn/en).

### CODE REVIEW

**Comment by gutjuri** [comment]
Thank you very much for the PR! I am glad to merge this when I get around to testing it.

I noticed that you use a pre-build Docker image in the PR. However, I was not able to find the corresponding Dockerfile neither in secretflow's Github repo nor on Docker Hub. Could you give me a pointer on where it is? Is it this one? https://github.com/secretflow/secretflow/blob/main/docker/dev/Dockerfile

I wonder whether we should just copy the Dockerfile into this repo since we use a fixed version of secretflow anyways and offering the ability for users to build from source themselves adds a certain transparency. The other MPC frameworks are also built from source in their respective Dockerfiles.

**Comment by mkskeller** [comment]
From the tag I would guess they have used the following script: https://github.com/secretflow/secretflow/blob/main/docker/release/build.sh

I disagree on building from scratch. I think the other frameworks have to be built from scratch because there is no binary release available, expect for MP-SPDZ where the binary release is the default. I recently ran all the frameworks for a project, and I would much rather prefer the faster option if available. I don't think the average user of this repository is interested in a "farm-to-table" approach on binaries.

File: secretflow/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+WORKDIR /root
+
+ADD source ./
+
+RUN ./mult3.py 1 2 3
+RUN ./innerprod.py 10
+RUN ./xtabs.py 10
+RUN ./xtabs_jax.py 10
+RUN ./cmp.py 10
```
Comment (gutjuri) [inline secretflow/Dockerfile]:
```suggestion
```

**Review by gutjuri — COMMENTED** [review]
Okay, understandable. In the meantime I got around to testing your PR. Works beautifully. Thanks.

One more thing. Could you please remove the RUN statements from the Dockerfile? I think build time of the image is not optimal for also running the examples. We could, however, add a script (e.g. `run_all.sh`) that runs all examples at once if you would like to have such a thing.

**Comment by mkskeller** [comment]
I've commented them out. My idea for this was a combination of testing and documentation.

**Comment by gutjuri** [comment]
Okay, thanks for applying the changes. As this looks good-to-go to me, I'll merge.

**Pull request merged**